pom.xml

```diff
@@ -1743,6 +1742,19 @@
           </dependency>
         </dependencies>
       </dependencyManagement>
+      <dependencies>
+        <!-- Remove log4j-slf4j-impl dependency -->
```

Comment:
Even with this configuration, `log4j-slf4j-impl` will still be in the dependency tree, but its scope is provided. Do you have any suggestions? @jerqi

Reply:
Both is ok for me. If we reserve the configuration, we should add some comments to explain it.

Reply:
Will it cause any trouble if we have both `log4j-slf4j-impl` and `log4j-slf4j2-impl` the same time in our client/server jars?

Reply:
The scope `provided` ensures that our final packaged artifact(the client and server jars) does not include this dependency, but it will be present during the `compile` and `test` phases. Will this cause any issues? @LuciferYang @jerqi

Reply:
![image](https://github.com/apache/incubator-uniffle/assets/1475305/b37e3c22-37b7-4972-86c8-fda17b3f37b3)

Yes, there are some classes with the same name under these two jars, which may cause some unknown problems due to the classpath loading order during testing. This is the risk I can think of